src/moscot/base/output.py

```diff
@@ -168,6 +171,98 @@ def chain(self, outputs: Iterable["BaseSolverOutput"], scale_by_marginals: bool
+        sparsified transport matrix stored as a :class:`~scipy.sparse.csr_matrix`.
+
+        Parameters
+        ----------
+        mode
+            How to determine the value below which entries are set to 0:
+
+                - 'threshold' - threshold below which entries are set to 0.0.
+                - 'percentile' - determine threshold by percentile below which entries are set to 0.0. Hence, between 0
+                  and 100.
+                - 'min_row' - choose the threshold such that each row has at least one non-zero entry.
+
+        value
+            Value to use for sparsification depending on ``mode``:
+
+                - `threshold` - `value` sets the threshold below which entries are set to 0.0.
+                - `percentile` - `value` is the percentile below which entries are set to 0.0.
+                - `min_row` - `value` is not used.
+        batch_size
+            How many rows of the transport matrix to sparsify per batch.
+        n_samples
+            If ``mode = 'percentile'``, determine the number of samples based on which the percentile
+            is computed stochastically. Note this means that a matrix of shape `[n_samples, transport_matrix.shape[1]]`
+            has to be instantiated. If `None`, ``n_samples`` is set to ``batch_size``.
+        seed
+            Random seed needed for sampling if ``mode = 'percentile'``.
+
+        Returns
+        -------
+        Returns :class:`moscot.base.MatrixSolverOutput` with a sparsified transport matrix.
+
+        Note
+        ----
+        This function only serves for interfacing software which has to instantiate the transport matrix. Methods in
+        :mod:`moscot` never use the sparsified transport matrix.
+        """
+        n, m = self.shape
+        if mode == "threshold":
+            if value is None:
+                raise ValueError("If `mode` is `threshold`, `threshold` must not be `None`.")
+            thr = value
+        elif mode == "percentile":
+            if value is None:
+                raise ValueError("If `mode` is `percentile`, `threshold` must not be `None`.")
+            rng = np.random.RandomState(seed=seed)
+            n_samples = n_samples if n_samples is not None else batch_size
+            k = min(n_samples, n)
+            x = np.zeros((m, k))
+            rows = rng.choice(m, size=k)
+            x[rows, np.arange(k)] = 1.0
+            res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+            thr = np.percentile(res, value)
+        elif mode == "min_row":
+            thr = np.inf
+            for batch in range(0, m, batch_size):
+                x = np.eye(m, min(batch_size, m - batch), -(min(batch, m)))
+                res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+                thr = min(thr, res.max(axis=1).min())
+        else:
+            raise NotImplementedError(mode)
+
+        if n < m:
+            k = n
```

Comment:
All these assignment can be done in 1 line to save space (also in the else branch).